exercises/grep/grep.spec.js

```diff
@@ -0,0 +1,371 @@
+    const stdout = child.stdout.toString().trim();
+
+    if (stderr) {
+      reject(stderr);
+    }
+
+    resolve(stdout);
+  });
+};
+
+const formatStringTemplate = stringTemplate =>
+  stringTemplate
+    .split("\n")
+    .map(sentence => sentence.trim())
+    .join("\n");
+
+describe("grep exercise", () => {
+  describe("Test grepping a single file", () => {
+    it("One file, one match, no flags", () => {
+      return expect(
+        spawn_grep({
+          pattern: "Agamemnon",
+          flags: [],
+          files: ["iliad.txt"]
+        })
+      ).resolves.toBe("Of Atreus, Agamemnon, King of men.");
+    });
+
+    it("One file, one match, print line numbers flag", () => {
+      return expect(
+        spawn_grep({
+          pattern: "Forbidden",
+          flags: ["-n"],
+          files: ["paradise-lost.txt"]
+        })
+      ).resolves.toBe("2:Of that Forbidden Tree, whose mortal tast");
+    });
+
+    it("One file, one match, case-insensitive flag", () => {
+      return expect(
+        spawn_grep({
+          pattern: "FORBIDDEN",
+          flags: ["-i"],
+          files: ["paradise-lost.txt"]
+        })
+      ).resolves.toBe("Of that Forbidden Tree, whose mortal tast");
+    });
+
+    it("One file, one match, print file names flag", () => {
+      return expect(
+        spawn_grep({
+          pattern: "Forbidden",
+          flags: ["-l"],
+          files: ["paradise-lost.txt"]
+        })
+      ).resolves.toBe(resolveDataFile("paradise-lost.txt"));
+    });
+
+    it("One file, one match, match entire lines flag", () => {
+      return expect(
+        spawn_grep({
+          pattern: "With loss of Eden, till one greater Man",
+          flags: ["-x"],
+          files: ["paradise-lost.txt"]
+        })
+      ).resolves.toBe("With loss of Eden, till one greater Man");
+    });
+
+    it("One file, one match, multiple flags", () => {
+      return expect(
+        spawn_grep({
+          pattern: "OF ATREUS, Agamemnon, KIng of MEN.",
+          flags: ["-n", "-i", "-x"],
+          files: ["iliad.txt"]
+        })
+      ).resolves.toBe("9:Of Atreus, Agamemnon, King of men.");
+    });
+
+    it("One file, several matches, no flags", () => {
+      return expect(
+        spawn_grep({
+          pattern: "may",
+          flags: [],
+          files: ["midsummer-night.txt"]
+        })
+      ).resolves.toBe(
+        formatStringTemplate(`Nor how it may concern my modesty,
+        But I beseech your grace that I may know
+        The worst that may befall me in this case,`)
+      );
+    });
+
+    it("One file, several matches, print line numbers flag", () => {
+      return expect(
+        spawn_grep({
+          pattern: "may",
+          flags: ["-n"],
+          files: ["midsummer-night.txt"]
+        })
+      ).resolves.toBe(
+        formatStringTemplate(
+          `3:Nor how it may concern my modesty,
+            5:But I beseech your grace that I may know
```

Comment:
Indentation looks inconsistent here to the `formatStringTemplates` used below

Reply:
Weird, I used prettier to format the file, I'll check it :+1:

Reply:
Ah, it might be, that you started the backtick in a new row then, the other ones, you put it into the same line as `formatStringTemplate`, which might be the reason for prettier to handle it that way.

Then disregard my comment :)